app/controllers/payment-links/get-amount-controller.js

```diff
@@ -0,0 +1,20 @@
+'use strict'
+
+// NPM dependencies
+const lodash = require('lodash')
+
+// Local dependencies
+const {response} = require('../../utils/response.js')
+const paths = require('../../paths')
+
+module.exports = (req, res) => {
+  const pageData = lodash.get(req, 'session.pageData.createPaymentLink', {})
+  const paymentLinkAmount = req.body['payment-amount'] || pageData.paymentLinkAmount || ''
```

Comment:
Should we not derive paymentAmountType from paymentLinkAmount and pass it to down to the template instead of having that logic in the template?